aten/src/ATen/native/nested/NestedTensorMath.cpp

```diff
@@ -178,7 +178,7 @@ std::tuple<Tensor, Tensor, Tensor> nested_layer_norm(
   Tensor mean = at::empty({M}, options);
   Tensor rstd = at::empty({M}, options);
   LayerNormKernel(
-      input_buffer.is_cuda() ? kCUDA : kCPU,
+      input_buffer.device().type(),
       input_buffer,
```

Comment:
```C++
if (input_buffer.is_cuda()) {
    auto acc_type = at::toAccumulateType(input_buffer.scalar_type(), true);
    options = options.dtype(acc_type);
  }
```
Is it necessary to include the XPU logic on line 174 as shown above?

Reply:
Thanks for the catch, yes. Updated.